Realm/Realm/Native/NativeCommon.cs

```diff
@@ -69,10 +69,17 @@ public static unsafe void Initialize()
 
                 if (platform.ToString() == "Win32NT")
                 {
-                    var assemblyLocation = Path.GetDirectoryName((string)assemblyLocationPI.GetValue(typeof(NativeCommon).GetTypeInfo().Assembly));
-                    var architecture = InteropConfig.Is64BitProcess ? "x64" : "x86";
-                    var path = Path.Combine(assemblyLocation, "lib", "win32", architecture) + Path.PathSeparator + Environment.GetEnvironmentVariable("PATH");
-                    Environment.SetEnvironmentVariable("PATH", path);
+                    try
+                    {
+                        var assemblyLocation = Path.GetDirectoryName((string)assemblyLocationPI.GetValue(typeof(NativeCommon).GetTypeInfo().Assembly));
+                        var architecture = InteropConfig.Is64BitProcess ? "x64" : "x86";
+                        var path = Path.Combine(assemblyLocation, "lib", "win32", architecture) + Path.PathSeparator + Environment.GetEnvironmentVariable("PATH");
+                        Environment.SetEnvironmentVariable("PATH", path);
+                    }
+                    catch (PlatformNotSupportedException)
```

Comment:
I was unable to figure out a way to determine whether we're UWP or regular Windows Desktop app :/ The hack with checking for a Windows Runtime type doesn't work, as you can reference WindowsRuntime in a regular app. Suggestions are welcome.